From 9185e2b96bf380f2d4f83dc6a7528651feea5775 Mon Sep 17 00:00:00 2001
From: Hendrik Brueckner <brueckner@redhat.com>
Date: Wed, 23 Jul 2014 10:49:50 -0400
Subject: [virt] tools/virtio: fix missing kmemleak_ignore symbol

Message-id: <1406112611-9862-3-git-send-email-brueckner@redhat.com>
Patchwork-id: 86317
O-Subject: [RHEL7.1 PATCH 02/23] [virt] tools/virtio: fix missing kmemleak_ignore symbol
Bugzilla: 1088554
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Description
-----------
commit 6abb2dd92868927c1d680546b9d13f208373d517
Author: Joel Stanley <joel@jms.id.au>
Date:   Thu Feb 13 15:03:46 2014 +1030

    tools/virtio: fix missing kmemleak_ignore symbol

    In commit bb478d8b167 virtio_ring: plug kmemleak false positive,
    kmemleak_ignore was introduced. This broke compilation of virtio_test:

      cc -g -O2 -Wall -I. -I ../../usr/include/ -Wno-pointer-sign
        -fno-strict-overflow -fno-strict-aliasing -fno-common -MMD
        -U_FORTIFY_SOURCE   -c -o virtio_ring.o ../../drivers/virtio/virtio_ring.c
      ../../drivers/virtio/virtio_ring.c: In function 'vring_add_indirect':
      ../../drivers/virtio/virtio_ring.c:177:2: warning: implicit declaration
      of function 'kmemleak_ignore' [-Wimplicit-function-declaration]
        kmemleak_ignore(desc);
        ^
      cc   virtio_test.o virtio_ring.o   -o virtio_test
      virtio_ring.o: In function `vring_add_indirect':
      tools/virtio/../../drivers/virtio/virtio_ring.c:177:
      undefined reference to `kmemleak_ignore'

    Add a dummy header for tools/virtio, and add #incldue <linux/kmemleak.h>
    to drivers/virtio/virtio_ring.c so it is picked up by the userspace
    tools.

    Signed-off-by: Joel Stanley <joel@jms.id.au>
    Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>

Bugzilla
--------
BZ 1088554
https://bugzilla.redhat.com/show_bug.cgi?id=1088554

Upstream status of the patch
----------------------------
The patch is upstream as of kernel version
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=6abb2dd92868927c1d680546b9d13f208373d517

Brew
----
https://brewweb.devel.redhat.com/taskinfo?taskID=7725759

Test status
-----------
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/virtio/virtio_ring.c b/drivers/virtio/virtio_ring.c
index 1e6f45d..753b2ab 100644
--- a/drivers/virtio/virtio_ring.c
+++ b/drivers/virtio/virtio_ring.c
@@ -23,6 +23,7 @@
 #include <linux/slab.h>
 #include <linux/module.h>
 #include <linux/hrtimer.h>
+#include <linux/kmemleak.h>
 
 #ifdef DEBUG
 /* For development, we want to crash whenever the ring is screwed. */
diff --git a/tools/virtio/linux/kmemleak.h b/tools/virtio/linux/kmemleak.h
new file mode 100644
index 0000000..c070722
--- /dev/null
+++ b/tools/virtio/linux/kmemleak.h
@@ -0,0 +1,3 @@
+static inline void kmemleak_ignore(const void *ptr)
+{
+}
-- 
1.7.1